extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1252,18 +1250,23 @@ export class AmpA4A extends AMP.BaseElement {
             'height': this.element.getAttribute('height'),
             'width': this.element.getAttribute('width'),
             'src': srcPath,
-            'name': nameData,
           }, SHARED_IFRAME_PROPERTIES));
-      if (method == XORIGIN_MODE.NAMEFRAME) {
         // TODO(bradfrizzell): change name of function and var
-        const attributes = getContextMetadata(
-            this.win, this.element, this.sentinel);
+      let attributes = getContextMetadata(
+          this.win, this.element, this.sentinel);
+      if (method == XORIGIN_MODE.NAMEFRAME) {
         attributes['creative'] = creative;
-        const name = JSON.stringify(attributes);
+        name = JSON.stringify(attributes);
         // Need to reassign the name once we've generated the context
         // attributes off of the iframe. Need the iframe to generate.
         iframe.setAttribute('name', name);
```

Comment:
Can move iframe.setAttribute calls outside if / else if by doing:

if (name) {
   iframe.setAttribute('name', name);
}
if (this.sentinel) {
  iframe.setAttribute('data-amp-3p-sentinel', this.sentinel);
}